file/pkg/service/service.go

```diff
@@ -80,3 +82,44 @@ func (f *fileService) ListFileShare(ctx context.Context, in *pb.ListFileShareReq
 	log.Debugf("Listed File Share successfully, #num=%d, fileshares: %+v\n", len(fileshares), fileshares)
 	return nil
 }
+
+func (f *fileService) GetFileShare(ctx context.Context, in *pb.GetFileShareRequest, out *pb.GetFileShareResponse) error {
+	log.Info("Received GetFileShare request.")
+
+	fs, err := db.DbAdapter.GetFileShare(ctx, in.Id)
+	if err != nil {
+		log.Errorf("failed to get fileshare: %v\n", err)
+		return err
+	}
+	var tags []*pb.Tag
+	for _, tag := range fs.Tags {
+		tags = append(tags, &pb.Tag{
+			Key:   tag.Key,
+			Value: tag.Value,
+		})
+	}
+	out.Fileshare = &pb.FileShare{
+		Id:                 fs.Id.Hex(),
```

Comment:
Is this ID auto-generated? What about the actual ID of the FS from cloud..Like AWS has their own FS-ID reported, which is used for further File ops...For Azure there is no ID returned but the ops can be done on Name

Reply:
Yes, This ID is auto generated to persist in mongo DB.
The backend specific ID are persisted as part of metadata.